src/fs.zig

```diff
@@ -67,7 +67,23 @@ pub fn serve(ctx: void, response: *Response, request: Request) ServeError!void {
     // and therefore has access to outside root.
     if (std.mem.startsWith(u8, path, "..")) return response.notFound();
 
-    const file = dir.openFile(path, .{}) catch |err| switch (err) {
+    // if the path is '', we should serve the index at root
+    const new_path = if (path.len == 0) index else blk: {
+        // if the path is a directory, we should serve the index inside
+        const stat = dir.statFile(path) catch |err| switch (err) {
+            error.FileNotFound => return response.notFound(),
+            else => |e| return e,
+        };
+        if (stat.kind == .Directory) {
+            break :blk try std.mem.concat(alloc, u8, &.{ path, "/", index });
```

Comment:
This leaks memory as the result is never freed. The best solution is to have a buffer with the size `std.os.MAX_PATH_BYTES` and use that buffer to backup the concatenation using memcpy. (similarly to line 52).